sys/fs/fuse/fuse_vnops.c

```diff
@@ -368,6 +368,84 @@ fuse_inval_buf_range(struct vnode *vp, off_t filesize, off_t start, off_t end)
 	return (0);
 }
 
+/* Send FUSE_IOCTL for this node */
+static int
+fuse_vnop_do_ioctl(struct vnode *vp, u_long cmd, void *arg, struct thread *td,
+	struct ucred *cred)
+{
+	struct fuse_dispatcher fdi;
+	struct fuse_ioctl_in *fii;
+	struct fuse_ioctl_out *fio;
+	struct fuse_filehandle *fufh;
+	uint32_t flags = 0;
+	uint32_t insize = 0;
+	uint32_t outsize = 0;
+	int err;
+
+	err = fuse_filehandle_get_anyflags(vp, &fufh, cred, td->td_proc->p_pid);
```

Comment:
Should this be fuse_filehandle_getrw instead?

Reply:
That's true for all _good_ fuse file systems.  But one (anti-, IMHO) feature of the fuse protocol is that the file system itself can choose to take responsibility for permission checks on open(). That's why fuse file handles exist.  So we need to be careful to get a file handle of the correct type.